thousandeyes/provider_test.go

```diff
@@ -0,0 +1,71 @@
+package thousandeyes
+
+import (
+	"context"
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	"github.com/thousandeyes/thousandeyes-sdk-go/v2"
+	"os"
+	"strconv"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+type ResourceType struct {
+	Name         string
+	ResourceName string
+	GetResource  func(id int64) (interface{}, error)
+}
+
+var testClient *thousandeyes.Client
+
+var providerFactories = map[string]func() (*schema.Provider, error){
+	"thousandeyes": func() (*schema.Provider, error) {
+		return New()(), nil
+	},
+}
+
+func init() {
+}
+
+func testAccPreCheck(t *testing.T) {
+	providerFunc, _ := providerFactories["thousandeyes"]
+	provider, err := providerFunc()
+	if err != nil {
+		fmt.Printf("Error creating provider: %v", err)
+		os.Exit(1)
+	}
+
+	ctx := context.TODO()
+	resourceData := schema.TestResourceDataRaw(t, provider.Schema, map[string]interface{}{})
+	testClientRaw, diags := provider.ConfigureContextFunc(ctx, resourceData)
+	if diags != nil && diags.HasError() {
+		fmt.Printf("Error configuring client: %v", diags)
```

Comment:
Should we use `require` here instead? It automatically calls `t.Fail()` if something doesn't pass the assertions:
https://pkg.go.dev/github.com/stretchr/testify/require

Reply:
sounds good! done